src/boot/AppEventHandlers.js

```diff
@@ -46,15 +66,28 @@ type Props = {|
 |};
 
 class AppEventHandlers extends PureComponent<Props> {
+  /** NetInfo disconnection mechanism.
+   *
+   * rn-netinfo has moved to a Boost `scoped_connection`- or .NET
+   * `IObservable`-style "return your disconnector" pattern, instead of using
+   * the hitherto-standard pair of `{add,remove}EventListener`.
+   *
+   * This is conceptually cleaner in some ways, and integrates much better with
+   * React's new "Hooks" mechanism... but Hooks don't integrate well with
+   * classes.
+   */
```

Comment:
Interesting bit of history! I'm not familiar with those two examples, though I feel like I've seen this "return your disconnector" pattern in some other environment before. Definitely agree that it's cleaner.

One other example within RN and our app is over in `notification/index.js` in the `NotificationListener#listen` method. The `DeviceEventEmitter` API provided by RN itself works essentially this way. In fact, a bit of grepping suggests that several (possibly all?) of the various libraries we add listeners to are using that API underneath, and just wrapping it as the less-clean pattern. Hopefully that means others will move to this one too in the future.

I think both these paragraphs would be good material for the commit message. I'd rather leave them out of the tree -- they feel more like commentary on the change from old to new than like something that explains the code as it now is.